src/storage/src/hummock/sstable/builder.rs

```diff
@@ -176,24 +176,42 @@ impl<W: SstableWriter, F: FilterBuilder> SstableBuilder<W, F> {
             total_key_count: 0,
             table_stats: Default::default(),
             last_table_stats: Default::default(),
+            range_tombstone_size: 0,
             epoch_set: BTreeSet::default(),
         }
     }
 
+    pub fn add_monotonic_deletes(&mut self, events: Vec<MonotonicDeleteEvent>) {
+        for event in events {
+            self.add_monotonic_delete(event);
+        }
+    }
+
     /// Add kv pair to sstable.
-    pub fn add_monotonic_deletes(&mut self, monotonic_deletes: Vec<MonotonicDeleteEvent>) {
-        let mut last_table_id = TableId::default();
-        for monotonic_delete in monotonic_deletes
-            .iter()
-            .filter(|monotonic_delete| monotonic_delete.new_epoch != HummockEpoch::MAX)
+    pub fn add_monotonic_delete(&mut self, event: MonotonicDeleteEvent) {
+        let table_id = event.event_key.left_user_key.table_id.table_id();
+        if self.last_table_id.is_none() || self.last_table_id.unwrap() != table_id {
```

Comment:
The `self.table_ids` seems useless in the builder. Can we just remove it?

Reply:
we need to record it in `SstableInfo`.